File: internal/systemd/discovery.go
package systemd

import (
	"fmt"
	"os"
)

// DiscoverServiceMode finds which mode a service is running in
func DiscoverServiceMode(name string) (SystemdMode, error) {
	// Check user mode first (new default)
	userPath := ServicePath(name, ModeUser)
	if _, err := os.Stat(userPath); err == nil {
		return ModeUser, nil
	}

	// Check system mode
	systemPath := ServicePath(name, ModeSystem)
	if _, err := os.Stat(systemPath); err == nil {
		return ModeSystem, nil
	}

	return 0, fmt.Errorf("service not found: %s", name)
}

File: src/systemd/discovery.rs
use crate::error::{Error, Result};
use crate::systemd::mode::SystemdMode;
use crate::systemd::paths::service_path;

pub fn discover_service_mode(name: &str) -> Result<SystemdMode> {
    if let Ok(p) = service_path(name, SystemdMode::User) {
        if p.exists() {
            return Ok(SystemdMode::User);
        }
    }
    if let Ok(p) = service_path(name, SystemdMode::System) {
        if p.exists() {
            return Ok(SystemdMode::System);
        }
    }
    Err(Error::msg(format!("service not found: {name}")))
}
